evalml/utils/woodwork_utils.py

```diff
@@ -132,3 +132,29 @@ def _convert_numeric_dataset_pandas(X, y):
         )
     y_ww = infer_feature_types(y)
     return X_ww, y_ww
+
+
+def _schema_is_equal(first, other):
+    """Loosely check whether or not the Woodwork schemas are equivalent. This only checks that the string values for the schemas are equal and doesn't take the actual type objects into account.
+
+    Args:
+        first (ww.Schema): The schema of the first woodwork datatable
+        other (ww.Schema): The schema of the second woodwork datatable
+
+    Returns:
+        bool: Whether or not the two schemas are equal
+    """
+    if first.types.index.tolist() != other.types.index.tolist():
+        return False
+    logical = [
+        x if x != "Integer" else "Double"
+        for x in first.types["Logical Type"].astype(str).tolist()
+    ] == [
+        x if x != "Integer" else "Double"
+        for x in other.types["Logical Type"].astype(str).tolist()
+    ]
+    semantic = (
```

Comment:
I wonder if we should just do `first.semantic_tags == second.semantic_tags`.

I think it'll be better because:
1. No need to use the "Semantic Tag(s)" string
2. I think the semantic tags are stored as a set, so the order should not matter,  but in this implementation, I think it's getting converted to a list?

![image](https://user-images.githubusercontent.com/41651716/142026068-87f52e08-afbd-4044-a485-cea36d19f768.png)


vs

![image](https://user-images.githubusercontent.com/41651716/142026166-9a0e335c-1f9c-4c4d-9421-bd5dac573ebf.png)